source/core/NumericString.cpp

```diff
@@ -1496,7 +1496,10 @@ Boolean TypedScanString(SubString* inputString, IntIndex* endToken, const Format
         }
         in.AliasAssign(in.Begin(), in.Begin() + std::min(in.Length(), formatOptions->MinimumFieldWidth + leadingSpace));
     }
-    truncateInput.Append(&in);
+    if (in.Length() <= truncateInput.Capacity())
+        truncateInput.Append(&in);
+    else
+        truncateInput.Append(&in, truncateInput.Capacity());
```

Comment:
I think there is another issue with using the truncated input string. It looks like we are not counting for the fact that we are operating on the truncated string for adjusting "endPointer" after finishing the scan. When we get a string representing double or complex double (especially over the wire) that may be longer than 255 characters, we should calculate the endPointer correctly so that "offset past scan" and "remaining string" will be correct. I think we should use this opportunity to fix this issue.

I would write a test that uses an input string "75.123........with 275 digits of precision or so, 45.8" and try to scan using "%f" and also using "%f, %f" (we should scan 45.8 correctly for the second double output) and make sure all the output parameters have the correct expected values.